### PR TITLE
feat: cancel dashboard queries when changing filters

### PR DESCRIPTION
**Pull request by pauldambra**
covers hitting refresh while refreshing but not changing filters while refreshing

## Problem

Dashboards can have expensive insight queries. users may change filters or get frustrated/distracted and navigate away. We don't cancel the queries, so ClickHouse does all of the work.

## Changes

Adds insight query cancellation to dashboards

### TODO

- [x] verify this is really cancelling queries
- [ ] tidy it up 😊 

punt to a follow-up
- [ ] changing filters currently waits for active refresh to finish and then loads the new set

## How did you test this code?

🤷 I didn't yet 🤣

### CODE REVIEW

**Comment by pauldambra** [comment]
I'm still working on this. Needs more testing (so there are still console logs and things)

But this is mostly there so thought I'd for an early(ish) sense-check.

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 1]
```diff
@@ -913,7 +921,7 @@ export const dashboardLogic = kea<dashboardLogicType>({
             actions.resetInterval()
             actions.refreshAllDashboardItems({ action: 'refresh_manual' })
         },
-        refreshAllDashboardItems: async ({ tiles, action, initialLoad, dashboardQueryId }, breakpoint) => {
+        refreshAllDashboardItems: async ({ tiles, action, initialLoad, dashboardQueryId = uuid() }, breakpoint) => {
```
Comment (macobo) [inline frontend/src/scenes/dashboard/dashboardLogic.tsx:924]: Is the = evaluated as in python (only once) or as you'd expect?

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 2]
```diff
@@ -929,31 +937,41 @@ export const dashboardLogic = kea<dashboardLogicType>({
...
-            dashboardQueryId = dashboardQueryId ?? uuid()
+
             let refreshesFinished = 0
             let totalResponseBytes = 0
 
             // array of functions that reload each item
             const fetchItemFunctions = insights.map((insight) => async () => {
-                // :TODO: Support query cancellation and use this queryId in the actual query.
-                const queryId = `${dashboardQueryId}::${uuid()}`
+                const queryId = `dash-${dashboardQueryId}::${uuid()}`
```
Comment (macobo) [inline frontend/src/scenes/dashboard/dashboardLogic.tsx:963]: I don't think the KILL logic below works due to the `dash-` prefix here. The backend checks whether the prefix of the query_id matches and we're not adding the `dash-` when you call cancel.

Suggestion: remove this.

**Comment by macobo** [comment]
It feels that AbortController and the KILL backend route are begging to be coupled together somehow...

Perhaps we should have a `KillLogic` which receives the latest query id that got kicked off, supplies methodOptions and kills running queries via an action?

File: frontend/src/scenes/dashboard/dashboardLogic.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/scenes/dashboard/dashboardLogic.tsx:963]: > I think the KILL logic below doesn't works due to the dash- prefix here

Ah, dang, yes. I added this, tested it, and then moved what id I was cancelling on 🤦

File: frontend/src/scenes/dashboard/dashboardLogic.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/dashboard/dashboardLogic.tsx:924]: Fair to check 😅 

But, yep, not like Python https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#evaluated_at_call_time

**Comment by pauldambra** [comment]
This has been superceded by other work

https://github.com/PostHog/posthog/pull/13178 and https://github.com/PostHog/posthog/pull/13414

**Pull request closed without merging**